website/static/js/pages/project-settings-page.js

```diff
@@ -54,8 +54,73 @@ if ($('#wgrid').length) {
         });
     });
 }
+
+
+var institutionsViewModel = function() {
+    ko.punches.enableAll();
+    var self = this;
+    self.primaryInstitution = ko.observable('None');
+    self.institutionHref = ko.observable('');
+    self.availableInstitutions = ko.observable();
+    $osf.ajaxJSON(
+        'GET',
+        ctx.apiV2Prefix + 'users/' + ctx.currentUser.id + '/?embed=institutions',
+        {isCors: true}
+    ).done(function(response) {
+        self.availableInstitutions(response.data.embeds.institutions.data);
+    });
+    $osf.ajaxJSON(
+        'GET',
+        ctx.apiV2Prefix + 'nodes/' + ctx.node.id + '/?embed=primary_institution',
+        {isCors: true}
+    ).done(function(response) {
+        if (response.data.embeds.primary_institution.data){
+            self.primaryInstitution(response.data.embeds.primary_institution.data.attributes.name);
+            self.institutionHref(response.data.embeds.primary_institution.data.links.html);
+        }
+    }).fail(function(){});
+    self.submitInst = function() {
+        var inst = $('input[name=primaryInst]:checked', '#selectedInst').val();
+        $osf.ajaxJSON(
```

Comment:
Any method that does async work should return a promise. In this case, you can return the return value of ajaxJSON.